### PR TITLE
feat: Point In Time Recovery (PITR)

### PR DESCRIPTION
**Pull request by skuruppu**

### CODE REVIEW

File: samples/database-create-with-version-retention-period.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    projectId: projectId,
+  });
+
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  try {
+    // Creates a new database.
+    console.log(`Creating database ${instance.formattedName_}.`);
+    const versionRetentionStatement = `
```
Comment (olavloite) [inline samples/database-create-with-version-retention-period.js:44]: nit: I think we should add a short comment here on the fact that the `database.create()` method does not support specifying a retention period, but that we execute it as an extra `ALTER DATABASE` statement after the database has been created. (I had to read the code a couple of times before I really saw that, as the sample and function is called `createDatabaseWithVersionRetentionPeriod`).

**Review by olavloite — APPROVED** [review]

File: samples/database-create-with-version-retention-period.js (hunk 1 above)
Comment (skuruppu) [reply samples/database-create-with-version-retention-period.js:45]: Yes, completely agree that this is not obvious. I added a comment to describe what's happening. Thanks for the suggestion.

**Review by thiagotnunes — APPROVED** [review]
LGTM

**Comment by skuruppu** [comment]
PTAL @thiagotnunes, I updated the backup/restore samples to include `version_time`.

**Review by thiagotnunes — APPROVED** [review]
LGTM

File: src/backup.ts [hunk 2]
```diff
@@ -178,6 +181,10 @@ class Backup {
         name: this.formattedName_,
       },
     };
+    if ('versionTime' in options) {
+      reqOpts.backup!.versionTime =
+        Spanner.timestamp(options.versionTime).toStruct();
+    }
```
Comment (skuruppu) [inline src/backup.ts:187]: PTAL this file since I had to add hand-written code to support `versionTime` when creating a backup.

**Review by thiagotnunes — APPROVED** [review]

File: src/backup.ts (hunk 2 above)
Comment (thiagotnunes) [reply src/backup.ts:187]: LGTM. Should we add `versionTime` to the `@example` in the comments?

Comment (skuruppu) [reply src/backup.ts]: Good idea. Done.

**Pull request merged**